source/write/replace.txt

```diff
@@ -0,0 +1,216 @@
+removes all fields and values from a specified document except the
+``_id`` field, and adds new fields and values that you specify. This
+operations differs from an update operation, which changes only
+specified fields in one or more documents.
+
+To learn more about update operations, see the
+:ref:`scala-write-update` guide.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To access this collection
+from your Scala application, create a ``MongoClient`` that connects to an Atlas cluster
+and assign the following value to your ``database`` and ``collection`` variables:
+
+.. literalinclude:: /includes/write/replace.scala
+    :language: scala
+    :dedent:
+    :start-after: start-db-coll
+    :end-before: end-db-coll
+
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+Replace Operation
+-----------------
+
+You can perform a replace operation in MongoDB by using the
+``replaceOne()`` method. This method removes all fields except the
+``_id`` field from the first document that matches the specified query filter. It
+then adds the fields and values you specify to the empty document.
+
+Required Parameters
+~~~~~~~~~~~~~~~~~~~
+
+You must pass the following parameters to the ``replaceOne()`` method:
+
+- **Query filter**, which matches which documents to update. To learn
+  more about query filters, see the :ref:`scala-specify-query`
+  guide.
+
+- **Replacement document**, which specifies the fields and values that
+  you want to replace the existing fields and values with.
+
+Replace One Document
+--------------------
+
+The following example uses the ``replaceOne()`` method to replace the
+fields and values of a document in which the value of the ``name`` field
+is ``"Primola Restaurant"``:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /includes/write/replace.scala
+      :start-after: start-replace-one
+      :end-before: end-replace-one
+      :language: scala
+      :dedent:
+
+   .. output::
+      :language: console
+      :visible: false
+
+      Replaced document count: 1
+      Completed
+
+.. important::
+
+   The values of ``_id`` fields are immutable. If your replacement
+   document specifies a value for the ``_id`` field, it must be the same
+   as the ``_id`` value of the existing document or the driver raises a
+   ``WriteError``.
+
+Customize the Replace Operation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``replaceOne()`` method optionally accepts
+a parameter that sets options to configure the replace operation.
+If you don't specify any options, the driver performs the replace
+operation with default settings.
+
+The following table describes the setter methods that you can use to
+configure an ``ReplaceOptions`` instance:
```

Comment:
S: I would mention the name of the parameter in the first paragraph so it's clear what the table is for. Would also avoid both "configure the replace operation" and "configure a ``ReplaceOptions`` instance. Maybe something similar to this:

```suggestion
The ``replaceOne()`` method optionally accepts
a parameter with the ``ReplaceOptions`` data type. The ``ReplaceOptions``
class contains setter methods that you can use to configure the replace operation.
If you don't specify any options, the driver performs the replace
operation with default settings.

The following table describes the setter methods in the ``ReplaceOptions``
class:
```